Sources/MapboxMaps/Foundation/Extensions/Core/DownloadStatus.swift

```diff
@@ -40,7 +40,7 @@ extension DownloadStatus {
                             downloadOptions: DownloadOptions,
                             httpResult: Result<HttpResponseData, HttpRequestError>?) {
 
-        let expected: Expected<AnyObject, AnyObject>?
+        let expected: Expected<HttpResponseData, HttpRequestError>?
```

Comment:
Is there a reason we need to combine the dependency version upgrade and adoption of lightweight generics? I would like to do some more thorough testing around lightweight generics before we adopt it, and I don't want to block the upgrade while we do that.

Reply:
After the update every single `Expected<AnyObject, AnyObject>` was causing a compilation error, e.g.:

> Cannot convert value of type '(Expected<AnyObject, AnyObject>?) -> Void' to expected argument type 'TileRegionGeometryCallback' (aka '(Expected<Geometry, TileRegionError>) -> ()')

As now Common requires `Expected` specialised with concrete types rather than `AnyObject`.